These functions are implemented in assembly for x86 and x86_64.
On other architectures, they are no-ops, and so parameters are unused.

Starting from version 3.3.7, mariadb-connector-c is compiled with "-Werror"
turned on, so allow unused parameters here with GCC's Diagnostic Pragma:
https://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html

--- a/libmariadb/ma_context.c
+++ b/libmariadb/ma_context.c
@@ -692,6 +692,9 @@
 #endif  /* MY_CONTEXT_USE_WIN32_FIBERS */
 
 #ifdef MY_CONTEXT_DISABLE
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
 int
 my_context_continue(struct my_context *c)
 {
@@ -723,4 +726,5 @@
 {
 }
 
+#pragma GCC diagnostic pop
 #endif
